plugins/modules/floating_ip.py

```diff
@@ -189,83 +221,68 @@
-
-        return resp
-
-    def update_info(self):
-        resp = self._get('floating-ips/' + self._module.params['ip'])
-        if resp:
-            self.info = self._resp2info(resp)
-        else:
-            self.info = {'ip': self._module.params['ip'],
-                         'state': 'absent'}
-
-    def request_floating_ip(self):
-        params = self._module.params
-
-        # check for required parameters to request a floating IP
-        self._module.fail_on_missing_params(('ip_version',))
-
-        data = {
-            'ip_version': params['ip_version'],
-            'server': params['server'],
-            'prefix_length': params['prefix_length'],
-            'reverse_ptr': params['reverse_ptr'],
-            'type': params['type'],
-            'region': params['region'],
-            'tags': params['tags'],
-        }
-
-        self.info = self._resp2info(self._post('floating-ips', data))
-
-    def release_floating_ip(self):
-        self._delete('floating-ips/%s' % self._module.params['ip'])
-        self.info = {'ip': self.info['ip'], 'state': 'absent'}
-
-    def update_floating_ip(self):
-        params = self._module.params
-
-        if not params['server']:
-            self._module.fail_json(msg='Missing required parameter to update a floating IP: server.')
-
-        data = {
-            'server': params['server'],
-            'tags': params['tags'],
-        }
-
-        self.info = self._resp2info(self._post('floating-ips/%s' % params['ip'], data))
+        super(AnsibleCloudscaleFloatingIp, self).__init__(
+            module=module,
+            resource_key_uuid='network',
+            resource_name='floating-ips',
+            resource_create_param_keys=[
+                'ip_version',
+                'server',
+                'prefix_length',
+                'reverse_ptr',
+                'type',
+                'region',
+                'tags',
+            ],
+            resource_update_param_keys=[
+                'server',
+                'reverse_ptr',
+                'tags',
+            ],
+        )
+        self.use_tag_for_name = True
+
+    def pre_transform(self, resource):
+        if 'server' in resource and isinstance(resource['server'], dict):
+            resource['server'] = resource['server']['uuid']
+        return resource
+
+    # TODO: To be removed in 2.0.0
+    def query(self):
+        # Don't try to find anything without network or name
+        if not any([self._module.params['name'], self._module.params['network']]):
+            return self.init_resource()
+        return super(AnsibleCloudscaleFloatingIp, self).query()
+
+    def create(self, resource):
+        # Fail when missing params for creation
+        self._module.fail_on_missing_params(['ip_version'])
+
+        # TODO: To be removed in 2.0.0
+        if not self._module.params['name']:
+            self._module.deprecate(
+                msg="Name is missing, this won't be supported in the future.",
```

Comment:
```suggestion
                msg="Name is missing, this won't be supported in the future",
```
IIRC deprecate adds a period at the end anyway. If you include one here, you'll get a double period.